src/components/Handlers/Handlers.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withModules } from '../Modules';
+import { stripesShape } from '../../Stripes';
+
+class Handlers extends React.Component {
+  static propTypes = {
+    stripes: stripesShape.isRequired,
+    event: PropTypes.number,
+    data: PropTypes.object,
+    modules: PropTypes.shape({
+      handler: PropTypes.array,
+    }),
+  };
+
+  constructor(props) {
+    super(props);
+
+    const { event, stripes, modules, data } = props;
+
+    this.components = modules.handler.reduce((acc, m) => {
+      const module = m.getModule();
+      const eventHander = module[m.handlerName];
+      if (!eventHander) return acc;
+      const component = eventHander(event, stripes, data);
+      if (component) {
+        acc.push(stripes.connect(component));
+      }
+      return acc;
+    }, []);
```

Comment:
(Side-note: it always seems stupid to me that the arguments to `reduce` are this way around. What it means is that every single use of it has a long-ass closure, and then dangling off the end of it, a vestigial `[]` or `{}`. It's most enough to make one define `Array.prototype.ecuder = (acc, fn) => reduce(fn, acc)`.)

Reply:
Yes I agree. `reduce` always looks ugly even when you try to make it look pretty. Ruby has a couple nice versions which I always liked: 
https://ruby-doc.org/core-2.1.0/Enumerable.html#method-i-reduce

Reply:
Ruby is, without a doubt, the cleanest language I've used extensively. It's not just that you can say so many things in it, it's that they all feel natural. The bitter but honest truth is that I never really felt comfortable around closures until I realised I'd been using them all the time in Ruby without even realising it: see https://reprog.wordpress.com/2010/02/27/closures-finally-explained/ if interested.